ci/build_setup.sh

```diff
@@ -129,7 +131,7 @@ mkdir -p "${ENVOY_BUILD_PROFILE}"
 mkdir -p "${ENVOY_FILTER_EXAMPLE_SRCDIR}"/bazel
 ln -sf "${ENVOY_SRCDIR}"/bazel/get_workspace_status "${ENVOY_FILTER_EXAMPLE_SRCDIR}"/bazel/
 cp -f "${ENVOY_SRCDIR}"/.bazelrc "${ENVOY_FILTER_EXAMPLE_SRCDIR}"/
-cp -f "${ENVOY_SRCDIR}"/*.bazelrc "${ENVOY_FILTER_EXAMPLE_SRCDIR}"/
+cp -f "$(bazel info workspace)"/*.bazelrc "${ENVOY_FILTER_EXAMPLE_SRCDIR}"/
```

Comment:
What's the reason for this line? You shouldn't touch `.bazelrc` anyway?

Reply:
`clang.bazel` is written to workspace (`bazel/setup_clang.sh`). This change follows that.

Reply:
But I don't think you want to build `envoy-filter-example` anyway in your use case?

Reply:
My use case is `envoy-openssl`. Without this change the script fails because `"${ENVOY_SRCDIR}"/*.bazelrc` matches no files (`clang.bazelrc` is elsewhere).

Reply:
What do you set for `ENVOY_FILTER_EXAMPLE_SRCDIR`? You don't build filter-example, do you?

Reply:
It gets built and tested as part of the existing Envoy build targets (e.g., `bazel.asan` and `bazel.tsan`) and works with the default `$ENVOY_FILTER_EXAMPLE_SRCDIR` setting.

Reply:
Sorry I'm confused what you're trying to do in the whole PR. Weren't you try to build your derivative? Why do you want to build envoy-filter-example in that case? Basically Envoy filter example is another derivative, so it should be a special case here.

Reply:
I want to re-use `ci/*` scripts in `envoy-openssl` CI but the same approach works with envoy-filter-example repository too (which I used as an example in the commit message).

> Why do you want to build envoy-filter-example in that case?

As I wrote it gets built and tested part of the existing Envoy build targets and it just gets inherited part of the 'reuse'.

Reply:
So are you trying to build `envoy-openssl` with `envoy-filter-example`? I guess no? So why do you   care about that? 

> As I wrote it gets built and tested part of the existing Envoy build targets and it just gets inherited part of the 'reuse'.

That's what I meant 'special case', you shouldn't build envoy-filter-example when you 'reuse' the script with `envoy-boringssl`.

In other word, can we refactor current `ENVOY_FILTER_EXAMPLE` build part into your derivative build process? That way the new part get tested regularly in CI.

Reply:
I'm OK to refactor [the filter-example build&test code](https://github.com/envoyproxy/envoy/blob/master/ci/do_ci.sh#L152:L156) out. I'd suggest another PR for that.

This PR does not bring any functional changes to existing Envoy CI builds but enables the reuse for `envoy-openssl` (and `envoy-filter-example`) repos. That's why the refactoring and functional changes could go in its own PR.

Is there something in this PR you want to see done differently?

Reply:
@lizan v2 pushed.